From 762e2e612445ed71ea933be429a6f01be950299c Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Thu, 19 Jun 2025 13:23:55 +0300
Subject: [PATCH] [SW-232783] lfs: g3: create itb file to clean lfs logs

Create itb file to cleanup LFS based logs as part
of build process

Change-Id: I6a9cbfdb5c6781789641aa7711323ee115bcfdb6
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/scripts/make/gaudi3/Makefile b/scripts/make/gaudi3/Makefile
index a2bd5ff..33918cc 100644
--- a/scripts/make/gaudi3/Makefile
+++ b/scripts/make/gaudi3/Makefile
@@ -179,7 +179,8 @@
 				gaudi3-slt-set-create				\
 				gaudi3-fake-fuse-create				\
 				gaudi3-rr-dump-create                           \
-				gaudi3-copy-efuse-to-fake-efuse-create
+				gaudi3-copy-efuse-to-fake-efuse-create		\
+				gaudi3-cleanup-log-files-create
 
 	@$(call hl-check-files,$(GAUDI3_INTERNAL_FITS))
 	@tar czf $(IMAGES_DIR)/${DEV_PREFIX}-int-fit.tgz -C $(ITBS_DIR)/ $(GAUDI3_INTERNAL_FITS)
diff --git a/scripts/make/hl-common/Makefile b/scripts/make/hl-common/Makefile
index 9e5b9bf..b164d45 100644
--- a/scripts/make/hl-common/Makefile
+++ b/scripts/make/hl-common/Makefile
@@ -424,6 +424,11 @@
 		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh read_log fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader no-load-only); \
 		mv $(ITBS_DIR)/${DEV_PREFIX}-read_log.itb $(ITBS_DIR)/${DEV_PREFIX}-read-log-files.itb
 
+.PHONY: %-cleanup-log-files-create
+%-cleanup-log-files-create:
+	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
+		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh cleanup_log fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader no-load-only); \
+		mv $(ITBS_DIR)/${DEV_PREFIX}-cleanup_log.itb $(ITBS_DIR)/${DEV_PREFIX}-cleanup-log-files.itb
 
 DEVICE_BIN_FITS_VAR = $(shell echo $(call hl-device, $@) | tr a-z A-Z)_BIN_FITS
 HL_BIN_FITS = $(or $($(DEVICE_BIN_FITS_VAR)),$(call hl-device, $@)-agent-fw_loader-fit.itb)
